phalcon/mvc/model/behavior/blameable.zep

```diff
@@ -0,0 +1,138 @@
+/*
+ +------------------------------------------------------------------------+
+ | Phalcon Framework                                                      |
+ +------------------------------------------------------------------------+
+ | Copyright (c) 2011-2015 Phalcon Team (http://www.phalconphp.com)       |
+ +------------------------------------------------------------------------+
+ | This source file is subject to the New BSD License that is bundled     |
+ | with this package in the file docs/LICENSE.txt.                        |
+ |                                                                        |
+ | If you did not receive a copy of the license and are unable to         |
+ | obtain it through the world-wide-web, please send an email             |
+ | to license@phalconphp.com so we can send you a copy immediately.       |
+ +------------------------------------------------------------------------+
+ | Authors: Andres Gutierrez <andres@phalconphp.com>                      |
+ |          Eduar Carvajal <eduar@phalconphp.com>                         |
+ |          Vladimir Metelitsa <green.cat@me.com>                         |
+ +------------------------------------------------------------------------+
+ */
+
+namespace Phalcon\Mvc\Model\Behavior;
+
+use Phalcon\Mvc\Model\Behavior;
+use Phalcon\Mvc\Model\BehaviorInterface;
+use Phalcon\Mvc\ModelInterface;
+
+/**
+ * Phalcon\Mvc\Model\Behavior\Blameable
+ */
+class Blameable extends Behavior implements BehaviorInterface
+{
+
+    /**
+     * Listens for notifications from the models manager
+     */
+    public function notify(string! type, <ModelInterface> model)
+    {
+        //Fires logAfterCreate if the event is afterCreate
+        if (eventType == "afterCreate") {
+            return this->auditAfterCreate(model);
+        }
+
+        //Fires logAfterUpdate if the event is afterUpdate
+        if (eventType == "afterUpdate") {
+            return this->auditAfterUpdate(model);
+        }
+    }
+
+    /**
+     * Creates an Audit instance based on the current enviroment
+     */
+    private function createAudit(string! type, <ModelInterface> model) -> <Audit>
+    {
+        var session, request, audit;
+        
+        session = model->getDI()->getSession();
+        request = model->getDI()->getRequest();
+        
+        audit = new Audit();
```

Comment:
i think this class should be returned by DI because if not it's mandatory to create an Audit object.